### ISSUE TITLE
How to create a dataset suitable for easyfsl from Tensor or non-image format matrix files?

### ISSUE THREAD

**Issue by misagamisaga** [sicara/easy-few-shot-learning#138]
Hello, author.

Of course, what needs to be said in front is that this is a very good library, thank you very much for the your contribution, we really want to use this library well, so we will raise such questions.

According to our understanding, our idea is, we wonder if there can be a more detailed part in the teaching document, which can teach us how to build a dataset suitable for `EasyFSL`?

On the one hand, the method of importing from json has been shown, but if it can appear in README file, it may be clearer, after all, there are indeed some users who are not used to reading the source code directly, but prefer to start a new library from the document or README file.

On the other hand, the input we encounter may not necessarily be only in picture format, so how should we build a dataset suitable for `EasyFSL`? For example, we are doing a medical-related topic now, and the source file we use is `.dcm` format. If we rashly convert it to picture format, there is a risk of information loss. Considering that you use Pytorch framework, this should be achievable. We currently cannot get this part of the information directly from the tutorial, and reading the source code directly also has a high cost of understanding. We hope to learn from the document or README file.

Thank you again for the author’s contribution, we hope to get the author’s reply, thank you.

**Comment by misagamisaga** [comment]
Well, I explored it myself and it seems that this problem has been solved. I will leave my own solution, perhaps it can be used as a reference for others who are interested in this issue

---

Your `Dataset` object needs to meet the following requirements:
1. Meet the general torch dataset object requirements
- Inherit from the `Dataset` class in torch
- Have `__init__`, `__len__`, `__getitem__` three methods
- It seems that you can omit `super(MyData, self).__init__()`
2. The first value returned by `__getitem__` is a tensor type image, and dtype=float32
- When reading the image, it should be read as a `np.array` of `dtype='float32'`
- Note that the dataset’s `transform` must have the `ToTensor` method
3. The second value returned by `__getitem__` is an int type label
- Note that `dataset.label` should be a `list` of `int` values

> Supplement:
> - `__getitem__` can return multiple values, but we will only use image and label
> - If image and label are not in the first and second positions of `__getitem__` return, you can re-annotate them in the second and third parameters of `WrapFewShotDataset` respectively

After defining the dataset using the general method, you need to do two steps:

1. Use the `WrapFewShotDataset` method to process the dataset for Few-Shot tasks
2. Check if the final dataset has a `.get_labels()` method, if not, you need to use the following method to supplement

> Here is the example code on the training set, the test set is similar
> 
> ```python
> def get_labels_train():
>     return train_set_ori.train_labels.tolist()
> 
> train_set.get_labels = get_labels_train
> ```
> 
> I didn't use the `lambda` method here because it would cause an error in Windows systems
